driver/src/contracts/stablex_contract.rs

```diff
@@ -65,6 +72,16 @@ pub trait StableXContract {
     /// Retrieve the time remaining in the batch.
     fn get_current_auction_remaining_time(&self) -> Result<Duration>;
 
+    /// Retrieve one page of indexed auction data that is filtered on chain
+    /// to only include orders valid at the given batchId.
+    fn get_filtered_auction_data_paginated(
```

Comment:
Should this method take a token filter as well? Or a `Option<Vec<TokenId>>` that defaults to the whitelist filter?